pkg/apis/core/types.go

```diff
@@ -5581,6 +5581,10 @@ type NodeStatus struct {
 	// +featureGate=SupplementalGroupsPolicy
 	// +optional
 	Features *NodeFeatures
+	// DeclaredFeatures represents the declared features of a node.
+	// +featureGate=NodeDeclaredFeatures
+	// +optional
+	DeclaredFeatures []string
```

Comment:
It looks like you're missing validation (https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/5328-node-declared-features/README.md#declared-feature-semantics).

You also need to drop the field if the feature gate is disabled. See https://github.com/kubernetes/kubernetes/blob/af9fb799ef09bbdb0b2b40b4e441f2ffccaffe18/pkg/registry/core/node/strategy.go#L94

Reply:
Node validation: https://github.com/kubernetes/kubernetes/blob/af9fb799ef09bbdb0b2b40b4e441f2ffccaffe18/pkg/apis/core/validation/validation.go#L6918
Node update validation: https://github.com/kubernetes/kubernetes/blob/af9fb799ef09bbdb0b2b40b4e441f2ffccaffe18/pkg/apis/core/validation/validation.go#L6975

Reply:
Thanks. Added validation checks and unit test to validate all of the registered features.